src/pages/workshops/WorkshopSingle.js

```diff
@@ -0,0 +1,475 @@
+        console.log("editing");
+        const requestOptions = {
+            method: "PUT",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify({
+                _id: workshopID,
+                content: {
+                    title: updateWorkTitle,
+                    description: updateWorkDesc,
+                    archived: updateWorkStatus,
+                    hosts: updateWorkHosts.map((item) => {
+                        return item.value;
+                    }),
+                    date: updateDate,
+                },
+            }),
+        };
+        fetch("http://localhost:3000/workshops", requestOptions).then(
+            (response) => {
+                setEditMode(false);
+            }
+        );
+    };
+    const enterUpdateMode = () => {
+        setUpdateWorkDesc(workshop.description);
+        setUpdateWorkTitle(workshop.title);
+        setUpdateWorkStatus(workshop.archived);
+        setUpdateDate(new Date(workshop.date).toISOString().substring(0, 10));
+        setEditMode(true);
+    };
+    useEffect(() => {
+        console.log("here");
+        fetch("http://localhost:3000/workshops?_id=" + workshopID)
+            .then((response) => response.json())
+            .then((data) => {
+                data = data[0];
+                fetch("http://localhost:3000/users/users")
+                    .then((response2) => response2.json())
+                    .then((data2) => {
+                        let tempOptions = data2
+                            .filter((item) => item.firstName && item.lastName)
+                            .map((item) => {
+                                return {
+                                    value: item._id,
+                                    label: item.firstName + " " + item.lastName,
+                                };
+                            });
+                        if (updateWorkHosts) {
+                            setUpdateWorkHosts([]);
+                        } //because useEffect gets called twice
+                        let tempHosts = data.hosts.map((item) => {
+                            for (let k = 0; k < tempOptions.length; k++) {
+                                if (item == tempOptions[k].value) {
+                                    setUpdateWorkHosts((oldArray) => [
+                                        ...oldArray,
+                                        tempOptions[k],
+                                    ]);
+                                    return tempOptions[k].label;
+                                }
+                            }
+                        });
+                        console.log(tempHosts);
+                        if (tempHosts) {
+                            data.hosts = tempHosts;
+                        }
+                        setHostOptions(tempOptions);
+                        setWorkshop(data);
+                    });
+            });
```

Comment:
error handling; also consider optimizing this code or abstracting it - looks kinda complex haha